back/src/domain/generic/notifications/useCases/SendNotification.unit.test.ts

```diff
@@ -1,14 +1,63 @@
-import { TemplatedEmail, TemplatedSms } from "shared";
-import { InMemoryNotificationGateway } from "../../../../adapters/secondary/notificationGateway/InMemoryNotificationGateway";
+import {
+  expectPromiseToFailWith,
+  expectPromiseToFailWithError,
+  TemplatedEmail,
+  TemplatedSms,
+} from "shared";
+import { createInMemoryUow } from "../../../../adapters/primary/config/uowConfig";
+import { InMemoryNotificationRepository } from "../../../../adapters/secondary/InMemoryNotificationRepository";
+import { InMemoryUowPerformer } from "../../../../adapters/secondary/InMemoryUowPerformer";
+import {
+  InMemoryNotificationGateway,
+  sendSmsErrorPhoneNumber,
+} from "../../../../adapters/secondary/notificationGateway/InMemoryNotificationGateway";
 import { SendNotification } from "./SendNotification";
 
+const someDate = new Date("2023-01-01").toISOString();
+
 describe("SendNotification UseCase", () => {
   let notificationGateway: InMemoryNotificationGateway;
+  let notificationRepository: InMemoryNotificationRepository;
   let sendNotification: SendNotification;
 
   beforeEach(() => {
     notificationGateway = new InMemoryNotificationGateway();
-    sendNotification = new SendNotification(notificationGateway);
+    const uow = createInMemoryUow();
+    notificationRepository = uow.notificationRepository;
+    const uowPerformer = new InMemoryUowPerformer(uow);
+    sendNotification = new SendNotification(uowPerformer, notificationGateway);
+  });
+
+  it("should throw if notification not found", async () => {
+    const id = "notif-404";
+    const kind = "email";
+    await expectPromiseToFailWith(
+      sendNotification.execute({ id, kind }),
+      `Notification with id ${id} and kind ${kind} not found`,
+    );
+  });
+
+  it("when error occures when trying to send SMS", async () => {
```

Comment:
"when error occures when trying to send SMS" ?

Sinon tu peux mettre un describe "wrong paths" par dessus tes cas d'erreur